src/classes/RequestHandler.py

```diff
@@ -30,64 +38,72 @@ def get_table_connection(self):
+        else:
+            previous_resource = ddb_response[Constants.DDB_RESPONSE_ATTRIBUTE_NAME_ITEMS][0]
+            if Constants.DDB_FIELD_CREATED_DATE not in previous_resource:
+                raise ValueError(
+                    'Resource with identifier ' + modified_resource.resource_identifier + ' has no ' + Constants.DDB_FIELD_CREATED_DATE + ' in DB')
+            else:
+                ddb_response = self.table.put_item(
+                    Item={
+                        Constants.DDB_FIELD_RESOURCE_IDENTIFIER: modified_resource.resource_identifier,
+                        Constants.DDB_FIELD_MODIFIED_DATE: current_time,
+                        Constants.DDB_FIELD_CREATED_DATE: previous_resource[Constants.DDB_FIELD_CREATED_DATE],
+                        Constants.DDB_FIELD_METADATA: modified_resource.metadata,
+                        Constants.DDB_FIELD_FILES: modified_resource.files,
+                        Constants.DDB_FIELD_OWNER: modified_resource.owner
+                    }
+                )
+                return ddb_response
 
     def handler(self, event, context):
-        operation = json.loads(event['body']).get('operation')
-        resource = json.loads(event['body']).get('resource')
-
-        # print('Operation - ' + operation)
-        current_time = arrow.utcnow().isoformat().replace("+00:00", "Z")
-
-        if operation == 'INSERT':
-            generated_uuid = uuid.uuid4().__str__()
-            ddb_response = self.insert_resource(generated_uuid, current_time, resource)
-            return {
-                'statusCode': 201,
-                'body': json.dumps(ddb_response),
-                'headers': {'Content-Type': 'application/json'}
-            }
-        elif operation == 'MODIFY':
-            ddb_response = self.modify_resource(resource, current_time)
-            return {
-                'statusCode': 200,
-                'body': json.dumps(ddb_response),
-                'headers': {'Content-Type': 'application/json'}
-            }
+        if event is None or Constants.EVENT_BODY not in event:
+            return response(http.HTTPStatus.BAD_REQUEST, 'Insufficient parameters')
         else:
-            raise ValueError("Unknown operation")
-
-
-# Helper class to convert a DynamoDB item to JSON.
-class DecimalEncoder(json.JSONEncoder):
-    def default(self, o):
-        if isinstance(o, decimal.Decimal):
-            if abs(o) % 1 > 0:
-                return float(o)
+            body = json.loads(event[Constants.EVENT_BODY])
+            operation = body.get(Constants.JSON_ATTRIBUTE_NAME_OPERATION)
+            resource_dict_from_json = body.get(Constants.JSON_ATTRIBUTE_NAME_RESOURCE)
+
+            try:
+                resource = Resource.from_dict(resource_dict_from_json)
+            except TypeError as e:
+                return response(http.HTTPStatus.BAD_REQUEST, e.args[0])
+
+            current_time = arrow.utcnow().isoformat().replace('+00:00', 'Z')
```

Comment:
Why are we doing this replace?

Reply:
'2019-11-01T13:14:42.747215Z'

Reply:
'2019-11-01T13:14:42.747215Z' vs '2019-11-01T13:15:22.921610+00:00' what format do we want?